upup/models/cloudup/resources/addons/authentication.aws/k8s-1.12.yaml.template

```diff
@@ -1,3 +1,78 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: iamidentitymappings.iamauthenticator.k8s.aws
+spec:
+  group: iamauthenticator.k8s.aws
+  version: v1alpha1
+  scope: Cluster
+  names:
+    plural: iamidentitymappings
+    singular: iamidentitymapping
+    kind: IAMIdentityMapping
+    categories:
+    - all
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+          - arn
+          - username
+          properties:
+            arn:
+              type: string
+            username:
+              type: string
+            groups:
+              type: array
+              items:
+                type: string
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: aws-iam-authenticator
+rules:
+- apiGroups:
+  - iamauthenticator.k8s.aws
+  resources:
+  - "*"
+  verbs:
+  - "*"
```

Comment:
@rifelpet I know you just copied this from the example, however giving the authenticator all permissions could potentially be dangerous no? As in if an attacker got access to this pod, corrupted a binary or otherwise found a way to inject data at this endpoint we could potentially be allowing someone to create new credentials that an administrator may not necessarily be watching for.

Should we change the verbs for the CRD to just get/list/watch?

Reply:
hm you're right. I opened https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/287 but anticipate it being a while before that is merged. I'll update this too and if theres a problem upstream we can update kops in a followup PR.